lib/wallet.py

```diff
@@ -1682,13 +1693,24 @@ def pubkeys_to_redeem_script(self, pubkeys):
     def derive_pubkeys(self, c, i):
         return [k.derive_pubkey(c, i) for k in self.get_keystores()]
 
+    def decide_txin_type(self):
```

Comment:
Maybe it would be better to add this method to `AbstractWallet` instead of duplicating it.